### ISSUE TITLE
Error when compile project in CudaSolver.cu

### ISSUE THREAD

**Issue by geotyper** [vcg-uvic/viper#7]
In line 551 CudaSolver.cu
`thrust::copy(S.X.v.begin(), S.X.v.end(), state.v.begin());`

before in line 314..
```
 S.X.x = state.x;
 S.X.q = state.q;
 S.X.r = state.r;
```
As idea that v it is not used and defined.
If comment this line all works well:

![Screenshot at 2020-05-15 21-05-39](https://user-images.githubusercontent.com/30755433/82112793-1ea16a80-9759-11ea-83e2-e59c0cf929f2.png)

**Comment by drebain** [comment]
I'm a bit confused, is that a line you wrote? It's not present in the version in this repository.

If so, could you clarify what you're trying to do? If `v` here is meant to be velocity, then that won't work as the solver is [position based](https://matthias-research.github.io/pages/publications/posBasedDyn.pdf) and doesn't store velocity explicitly.

**Comment by geotyper** [comment]
I download project from https://github.com/tomerwei/viper, and think that it is fork of original repository. But now understand that it have some modifications in code. thanks for prompt answer.

**Comment by geotyper** [comment]
And small additional question:
If make pills with spheres (a,b) and pills (a,c) and give this spheres a,b,c different groupId, the spheres don't collide with together, not only pairs (a,b) and (a,c), but don't collide (b,c). But they collide with other spheres in world if they haven't pills with them. Is code have some flag that disable collide with spheres in pills? Sorry for my english

**Comment by drebain** [comment]
The way it is implemented now is that there is only collisions between pills, we don't do collision detection or resolution for individual spheres. The reason for this is that doing so would require handling three types of collision (sphere-sphere, pill-pill, and pill-sphere) and we didn't have enough individual spheres to justify it. The only place we had them was for the cannonballs, which we made from [pills with the same endpoint](https://github.com/vcg-uvic/viper/blob/2f25416385b0cf42c60e19ff787fe4a6a4c26223/Demo/OctopusComponent.h#L498). 

For filtering pill collisions, we are just setting the group ID of a pill to be [that of its first sphere](https://github.com/vcg-uvic/viper/blob/2f25416385b0cf42c60e19ff787fe4a6a4c26223/Viper/Scene.cu#L29) so if you have three spheres connected into two pills then only two of the sphere IDs are actually used. I don't recall why we did it this way, but I think there was an earlier version with different behaviour.

If you are trying to make a self-colliding chain or rope then the group ID collision filtering might not be ideal for you. I think you might be able to just change [the filtering method](https://github.com/vcg-uvic/viper/blob/2f25416385b0cf42c60e19ff787fe4a6a4c26223/Viper/CudaSolver.cu#L270) to only look at whether the two pills share any spheres, or to be more robust, filter based on whether they collide in the rest configuration.
